packages/components/src/chart-uplot/index.tsx

```diff
@@ -49,16 +49,126 @@ interface UplotChartProps {
 	data: uPlot.AlignedData;
 	options?: Partial< uPlot.Options >;
 	legendContainer?: React.RefObject< HTMLDivElement >;
+	solidFill?: boolean;
 }
 
 export default function UplotChart( {
 	data,
 	options: propOptions,
 	legendContainer,
+	solidFill = false,
 }: UplotChartProps ) {
 	const translate = useTranslate();
 	const uplot = useRef< uPlot | null >( null );
 	const uplotContainer = useRef( null );
+	const defaultFillColor = 'rgba(48, 87, 220, 0.075)';
+
+	const can = document.createElement( 'canvas' );
+	const ctx = can.getContext( '2d' );
+
+	const scaleGradient = (
```

Comment:
Can we refactor this into a hook? Like `useScaleGradient`?

If not, could we either memorize this function declaration or move the function declaration outside of the component function?